### PR TITLE
feat:  add metrics for trainer

### PR DESCRIPTION
**Pull request by MIchaelFU0403**
Add metrics for trainer
## Description
Define some mesurement metrics for trainer 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: trainer/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	}, []string{"model_type", "scheduler_id"})
+
+	TrainStartedFailureCount = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: types.MetricsNamespace,
+		Subsystem: types.TrainerMetricsName,
+		Name:      "training_started_failure_total",
+		Help:      "Counter of the number of failed of the training started.",
+	}, []string{"model_type", "scheduler_id"})
+
+	TrainFinishedCount = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (fcgxz2003) [inline trainer/metrics/metrics.go:49]: Remove

File: trainer/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		Namespace: types.MetricsNamespace,
+		Subsystem: types.TrainerMetricsName,
+		Name:      "training_finished_total",
+		Help:      "Counter of the number of the training finished.",
+	}, []string{"model_type", "scheduler_id"})
+
+	TrainFinishedFailureCount = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (fcgxz2003) [inline trainer/metrics/metrics.go:56]: Remove

File: trainer/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		Namespace: types.MetricsNamespace,
+		Subsystem: types.TrainerMetricsName,
+		Name:      "training_finished_failure_total",
+		Help:      "Counter of the number of failed of the training finished.",
+	}, []string{"model_type", "scheduler_id"})
+
+	UploadModelCount = promauto.NewCounter(prometheus.CounterOpts{
```
Comment (fcgxz2003) [inline trainer/metrics/metrics.go:63]: CreateModelCount

File: trainer/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		Namespace: types.MetricsNamespace,
+		Subsystem: types.TrainerMetricsName,
+		Name:      "upload_total",
+		Help:      "Counter of the number of the upload trained model.",
+	})
+
+	UploadModelFailureCount = promauto.NewCounter(prometheus.CounterOpts{
```
Comment (fcgxz2003) [inline trainer/metrics/metrics.go:70]: CreateModelFailureCount

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**